DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/ViewTreeRecordingContext.swift

```diff
@@ -47,16 +49,23 @@ internal extension ViewTreeRecordingContext {
             ///
             /// - Parameter viewController: The `UIViewController` for which to determine the `ViewControllerType`.
             internal init(_ viewController: UIViewController?) {
+                guard let viewController else {
+                    self = .other
+                    return
+                }
+
                 switch viewController {
                 case is UIAlertController:
                     self = .alert
                 case is UIActivityViewController:
                     self = .activity
                 case is SFSafariViewController:
                     self = .safari
-                case is AnyUIHostingViewController:
-                    self = .swiftUI
                 default:
+                    if FrameworkUtils.isSwiftUI(class: type(of: viewController)) {
+                        self = .swiftUI
+                        return
+                    }
                     self = .other
                 }
```

Comment:
/suggestion Lets replace the `switch` statement with a `if else` for readability 👍